### PR TITLE
lib/route: Remove redundant implementation

### PR DESCRIPTION
**Pull request by sirreal**
https://github.com/Automattic/wp-calypso/pull/21137 duplicated several implementations. Remove the duplicate implementations.

These implementations were contained in the index as well as in independent modules. Move them all to typed independent modules exported from the index.

## Testing
- This is moving and deuplicating code. If the app builds and e2e tests pass, that's likely sufficient testing.
- page-notifier was not imported in the application and the file was removed.

### CODE REVIEW

File: client/lib/route/normalize.ts
Comment (sgomes) [inline client/lib/route/normalize.ts]: I don't think this will work. Perhaps you meant:
```suggestion
import untrailingslashit from './untrailingslashit';
```
?

**Review by sgomes — CHANGES_REQUESTED** [review]

File: client/lib/route/normalize.ts
Comment (sirreal) [reply client/lib/route/normalize.ts]: We can import from the specific module like you suggest if you'd prefer that.

Comment (sgomes) [reply client/lib/route/normalize.ts]: I'd recommend depending on the sibling module directly, instead, as it would make the dependency graph a bit saner. It's probably still fine, though.

**Review by sgomes — APPROVED** [review]
LGTM after the import changes. I'll leave it to you whether you want to depend on the sibling module directly, or do a named import from the index. Either way is fine 👍

File: client/lib/route/normalize.ts
Comment (sirreal) [inline client/lib/route/normalize.ts]:
```suggestion
import untrailingslashit from './untrailingslashit';
```

**Pull request merged**
